src/java.desktop/share/classes/javax/swing/plaf/basic/BasicHTML.java

```diff
@@ -61,8 +61,24 @@ public BasicHTML() {}
      */
     public static View createHTMLView(JComponent c, String html) {
         BasicEditorKit kit = getFactory();
+        int beginIndex = html.indexOf("rgba(");
+        Color col = c.getForeground();
+        int alpha = col.getAlpha();
+        boolean hasAlpha = false;
+        if (beginIndex != -1) {
+            int endIndex = html.indexOf(")");
+            String[] color = html.substring(beginIndex + "rgba(".length(),
+                                               endIndex).split(",");
+            for (int i = 0; i < color.length; i++) {
+                color[i] = color[i].strip();
+            }
+            col = new Color(Integer.parseInt(color[0]), Integer.parseInt(color[1]),
+                            Integer.parseInt(color[2]), (int)(Float.parseFloat(color[3])*255));
```

Comment:
It seems OceanTheme in Metal L&F supports only rgb

`ColorUIResource CONTROL_TEXT_COLOR =  new PrintColorUIResource(0x333333, Color.BLACK)`

so even when Color(r,g,b,a) is used, it creates rgb color and not rgba 
so alpha is obtained and passed separately.

Reply:
I have already mentioned it here.

Reply:
But in what place we parse "rgb(...)"? it does not seem mentioned in this method and does not depend on the c.getForeground()?

Reply:
It is parsed in Parser.java#parseAttributeValue() which is being called from this method via kit.read().
It also parses rgba() and store it but JDK does not seem to support rgba() in CSS.java and also SwingUtilities2.displayPropertiesToCSS() which is being called prior to html parsing so alpha needs to be parsed separately.